hypergan/gan.py

```diff
@@ -1,317 +1,340 @@
+            elif graph_type == 'generator':
+                graph = self.graph.create_generator(x,y,f)
+            else:
+                raise Exception("Invalid graph type")
+
+        return self.graph
+
+    def setup_loader(self, format, directory, device, seconds=None,
+            bitrate=None, crop=False, width=None, height=None, channels=3):
+        with tf.device('/cpu:0'):
+            #TODO mp3 braken
+            if(format == 'mp3'):
+                return audio_loader.mp3_tensors_from_directory(
+                        directory,
+                        self.config['batch_size'],
+                        seconds=seconds,
+                        channels=channels,
+                        bitrate=bitrate,
+                        format=format)
+            else:
+                return image_loader.labelled_image_tensors_from_directory(
+                        directory,
+                        self.config['batch_size'], 
+                        channels=channels, 
+                        format=format,
+                        crop=crop,
+                        width=width,
+                        height=height)
+
+
+                
+
+    def init_session(self, device):
+        # Initialize tensorflow
+        with tf.device(device):
+            self.sess = tf.Session(config=tf.ConfigProto())
+
+    def run(self):
+        args = self.args
+        crop = args.crop
+        channels = int(args.size.split("x")[2])
+        width = int(args.size.split("x")[0])
+        height = int(args.size.split("x")[1])
+        loadedFromSave = False
+
+        print("[hypergan] Welcome back.  You are one of ", self.selector.count_configs(), " possible configurations.")
+
+        self.config = self.selector.random_config()
+        self.load_config(args.config)
+        self.config['dtype']=tf.float32 #TODO fix.  this happens because dtype is stored as an enum
+
+        self.init_session(args.device)
+
+        self.config['batch_size']=args.batch_size
+        x,y,f,num_labels,examples_per_epoch = self.setup_loader(
+                args.format,
+                args.directory,
+                args.device,
+                seconds=None,
+                bitrate=None,
+                width=width,
+                height=height,
+                channels=channels,
+                crop=crop
+        )
+        self.config['y_dims']=num_labels
+        self.config['x_dims']=[height,width] #todo can we remove this?
+        self.config['channels']=channels
+        config = self.config
+
+        if args.config is None:
+            filename = '~/.hypergan/configs/'+config['uuid']+'.json'
+            print("[hypergan] saving network configuration to: " + filename)
+            config = self.selector.load_or_create_config(filename, config)
+        else:
+            save_file = "~/.hypergan/saves/"+args.config+".ckpt"
+            config['uuid'] = args.config
+
+        save_file = "~/.hypergan/saves/"+config["uuid"]+".ckpt"
```

Comment:
Suggestion: Remove this uuid stuff - that should come from the cli I think.  Lets require networks to be named.